pkg/flare/manifests.go

```diff
@@ -0,0 +1,206 @@
+	}
+	return &secretList.Items[0], nil
+}
+
+// Retrieve Helm chart user values from the API server for a given name and namespace looking through Secrets or Configmaps
+func GetDatadogHelmUserValues(cl *apiserver.APIClient, name string, namespace string, storage string) ([]byte, error) {
+	switch storage {
+	case "secret":
+		secret, err := getDeployedHelmSecret(cl, name, namespace)
+		if err != nil {
+			return nil, err
+		}
+		b64EncodedGzipRelease := secret.Data["release"]
+		// Contrary to the Configmap, the secret data is a byte array, so the string function is necessary
+		b, err := decodeChartValuesFromRelease(string(b64EncodedGzipRelease))
+		if err != nil {
+			return nil, err
+		}
+		return b, nil
+	case "configmap":
+		configmap, err := getDeployedHelmConfigmap(cl, name, namespace)
+		if err != nil {
+			return nil, err
+		}
+		b64EncodedGzipRelease := configmap.Data["release"]
+		b, err := decodeChartValuesFromRelease(b64EncodedGzipRelease)
+		if err != nil {
+			return nil, err
+		}
+		return b, nil
+	default:
+		return nil, log.Errorf("The requested storage %s is not supported", storage)
+	}
+}
+
+// decodeRelease decodes the bytes of data into a readable byte array.
+// Data must contain a base64 encoded gzipped string of a valid release, otherwise nil is returned.
+func decodeRelease(data string) ([]byte, error) {
+	// base64 decode string
+	b, err := b64.DecodeString(data)
+	if err != nil {
+		return nil, err
+	}
+
+	// For backwards compatibility with releases that were stored before
+	// compression was introduced we skip decompression if the
+	// gzip magic header is not found
+	if len(b) < 4 {
+		// Avoid panic if b[0:3] cannot be accessed
+		return nil, log.Errorf("The byte array is too short (expected at least 4 characters, got %s instead): it cannot contain a Helm release", fmt.Sprint(len(b)))
+	}
+	if bytes.Equal(b[0:3], magicGzip) {
+		r, err := gzip.NewReader(bytes.NewReader(b))
+		if err != nil {
+			return nil, err
+		}
+		defer r.Close()
+		b2, err := io.ReadAll(r)
+		if err != nil {
+			return nil, err
+		}
+		b = b2
+	}
+	return b, nil
+}
+
+// decodeChartValuesFromRelease returns a byte array with the user values from an encoded Helm chart release
+func decodeChartValuesFromRelease(encodedRelease string) ([]byte, error) {
+	var b bytes.Buffer
+	var userConfig chartUserValues
+
+	decodedrelease, err := decodeRelease(encodedRelease)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(decodedrelease, &userConfig)
+	if err != nil {
+		log.Debugf("Unable to retrieve the config data: %s", err.Error())
+		return nil, err
+	}
+	configjson, err := json.Marshal(userConfig)
+	if err != nil {
+		log.Debugf("Can't marshall user values into a proper JSON: %s", err.Error())
+		return nil, err
+	}
+	configyaml, err := yaml.JSONToYAML(configjson)
+	if err != nil {
+		log.Debugf("Can't convert user values to YAML: %s", err.Error())
+		return nil, err
+	}
+	fmt.Fprintln(&b, string(configyaml))
+	return b.Bytes(), nil
```

Comment:
for simplicity this can be replace by

```suggestion
	return yaml.JSONToYAML(configjson)
```